docs/src/references/c8y-firmware-management.md

```diff
@@ -0,0 +1,214 @@
+as this plugin doesn't support firmware updates for the tedge device.
+Operation files for child devices must be created as part of their bootstrap process, which is explained later.
+
+## Configuration
+
+The plugin supports a single tedge configuration named `firmware.child.update.timeout`,
+that defines the amount of time the plugin wait for a child device to finish a firmware update once the request is delivered.
+The default timeout value is `3600s` and can be updated with:
+
+```shell
+sudo tedge config set firmware.child.update.timeout <another-value>
+```
+
+## Usage
+
+```shell
+c8y-firmware-plugin --help
+```
+
+```shell
+<!-- cmdrun c8y-firmware-plugin --help -->
+```
+
+The `c8y-firmware-plugin` has to be run as a daemon on the device.
+On systemd supported OSes, it can be run as a daemon service as follows:
+
+```shell
+systemctl start c8y-firmware-plugin
+systemctl enable c8y-firmware-plugin
+```
+
+## Firmware update protocol between thin-edge and the child-devices
+
+The plugin manages the download and delivery of firmware files for child-devices connected to the thin-edge device,
+acting as a proxy between the cloud and the child-devices.
+The firmware updates are downloaded from the cloud on the thin-edge device then made available to the child-devices over HTTP,
+using MQTT to notify the availability of these firmware updates.
+The child-device software has to subscribe to these MQTT messages, download the corresponding updates via HTTP,
+and notify the firmware plugin of the update status via MQTT.
+
+* The responsibilities of the plugin are:
+  * to download the firmware files pushed from the cloud, caching it to be shared with child devices
+  * to handle network failures during the download even on flaky networks
+  * to publish the downloaded firmware files over a local HTTP server and make them available to the child-devices,
+  * to notify the child-devices when firmware updates are available,
+  * to receive forward the firmware update status updates from the child devices to the cloud
+* By contrast, the plugin is not responsible for:
+  * checking the integrity of the downloaded file which is a third-party binary
+  * installing the firmware files on the child-devices.
+* For each child-device, a device-specific software component, referred to as a `child-device-connector`, 
+  is required to listen for firmware update related MQTT notifications from the plugin
+  and behave accordingly along the protocol defined by this plugin.
+  * Being specific to each type of child devices, this software has to be implemented specifically.
+  * This software can be installed on the child device.
+  * This software can also be installed on the main device,
+    when the target device cannot be altered or connected to the main device over MQTT and HTTP.
+
+### Child device connector connecting to thin-edge device
+
+The `child-device-connector` interacts with thin-edge over its MQTT and HTTP APIs.
```

Comment:
A sentence about what the `child-device-connector` is would be useful to give it more context in case if user's didn't see the first reference to it in the list of responsibilities.

So maybe add something like this as the first sentence of the paragraph.

```markdown
The `child-device-connector` has the responsibility of handling firmware requests sent by the thin-edge and translates it to the relevant 3rd-party API to apply the contents of the firmware update request.
```